crates/holochain/src/conductor/conductor/tests/agent_key_revocation.rs

```diff
@@ -906,3 +911,60 @@ async fn revoke_agent_key_in_dpki(conductor: &SweetConductor, agent_key: AgentPu
         _state => panic!("key must be valid but is {_state:?}"),
     }
 }
+
+async fn delete_agent_key_from_source_chain(
+    conductor: &SweetConductor,
+    source_chain: &mut SourceChain,
+    cell_id: &CellId,
+) {
+    source_chain.delete_valid_agent_pub_key().await.unwrap();
+    source_chain
+        .flush(
+            &conductor
+                .holochain_p2p()
+                .to_dna(cell_id.dna_hash().clone(), conductor.get_chc(cell_id)),
+        )
+        .await
+        .unwrap();
+    conductor
+        .get_cell_triggers(cell_id)
+        .await
+        .unwrap()
+        .publish_dht_ops
+        .trigger(&"key deletion");
+    conductor
+        .get_cell_triggers(cell_id)
+        .await
+        .unwrap()
+        .integrate_dht_ops
+        .trigger(&"key deletion");
+}
+
+fn assert_delete_agent_key_accepted_by_validation(
+    agent_key: &AgentPubKey,
+    conductor: &SweetConductor,
+    dna_hash: &DnaHash,
+) {
+    let sql = "\
+        SELECT Action.author, Action.type, Action.deletes_entry_hash
+        FROM Action
+        JOIN DhtOp On DhtOp.action_hash = Action.hash
+        WHERE DhtOp.validation_status = 0
+        AND Action.deletes_entry_hash IS NOT NULL
+        ORDER BY seq DESC";
+    conductor.get_dht_db(dna_hash).unwrap().test_read({
+        let agent_key = agent_key.clone();
+        move |txn| {
+            let mut stmt = txn.prepare(sql).unwrap();
+            let mut rows = stmt.query([]).unwrap();
```

Comment:
Will `.query()` error if no rows are returned? If so, then this is good, if not, then that needs to be checked as well.

Reply:
Yes, it'll return [Error::QueryReturnedNoRows](https://docs.rs/rusqlite/latest/rusqlite/enum.Error.html#variant.QueryReturnedNoRows)

Reply:
@jost-s I'm not sure. The docs for [query](https://docs.rs/rusqlite/latest/rusqlite/struct.Statement.html#method.query) don't mention returning an error, but for [query_row](https://docs.rs/rusqlite/latest/rusqlite/struct.Statement.html#method.query_row) that error is mentioned, so this is either an inconsistency in rusqlite docs, or `query` does not actually expect rows to be returned.

Reply:
I've tested it anyway and it returns an error.

Reply:
I've added an assertion that some ops have been validated.